### PR TITLE
Everywhere: Replace LibSQL with sqlite3

### PR DESCRIPTION
**Pull request by trflynn89**
This is a bit of an experiment in handling vendored dependencies.

LibSQL was a home-grown database engine written for SerenityOS. It has had many issues over the years, and is still an incomplete engine with no ACID guarantees, and requires being run in an isolated, singleton process. By comparison, sqlite3 is a tried-and-true engine that is pretty easy to plug in and use.

This PR hooks vcpkg into ladybird.sh. Hopefully, anyone using this script will have to do exactly nothing to pick up these changes*

:warning: * Almost nothing. Removing the build directory is unfortunately required before using these changes, as we have to set some CMake toolchain info that doesn't like being changed in an existing build. :warning: 

~~vcpkg is theoretically already installed on the CI runners, so hopefully that also just works :tm:~~
edit: not quite true :(

This does not integrate with the GN build yet. That build is already quite out of date, so I plan to tackle that separately.

Screenshot_20240604_141310

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 cmake_minimum_required(VERSION 3.25)
 
+SET(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
```
Comment (ADKaster) [inline CMakeLists.txt:3]: setting the toolchain file in CMakeLists.txt is quite uncomfortable :(

Comment (trflynn89) [reply CMakeLists.txt:3]: From what I read, it's either this, or everyone *has* to use some `cmake --preset="preset with this toolchain"` themselves on the command line

**Comment by joekohlsdorf** [comment]
What do you think about enabling WAL mode? https://www.sqlite.org/wal.html
It might be beneficial when multiple Ladybird processes are running.

**Comment by trflynn89** [comment]
> What do you think about enabling WAL mode? [sqlite.org/wal.html](https://www.sqlite.org/wal.html) It might be beneficial when multiple Ladybird processes are running.

Unless we see performance regressions immediately, I'd rather handle that in future iterations.

(Also we have the main UI process, which owns the database, set up such that there is only ever one of them - if you launch a second, you get a new window/tab in the original process)

**Comment by elcuco** [comment]
May I propose another alternative to pulling 3rd parties? You can use CPM - https://github.com/cpm-cmake/CPM.cmake

This requires no setup from the developer. The downside - this does not handle dependencies. Meaning - if we want to add another library, that requires a 3rd one - we will need to add it manually. I did this for NotepadNext, and this was quite frictionless: see https://github.com/elcuco/NotepadNext/blob/cmake-build/CMakeLists.txt

**Comment by vitalyster** [comment]
Please don't overcomplicate things at this time. CMake-provided FindSQLite3.cmake module perfectly finds sqlite3 installed on Linux and Mac(Homebrew, Macports, etc.). Users on unsupported systems will be able to provide paths to installed library via CMake command line arguments.

**Comment by trflynn89** [comment]
> Please don't overcomplicate things at this time. CMake-provided FindSQLite3.cmake module perfectly finds sqlite3 installed on Linux and Mac(Homebrew, Macports, etc.). Users on unsupported systems will be able to provide paths to installed library via CMake command line arguments.

We (the maintainers of this project) are simply experimenting with different ideas at this time. We are fully aware that FindSQLite3, FetchContent, etc. exist. We already have plans for dependencies that do not have a FindFoo module, and several of us are just exploring our options.

**Review by ADKaster — APPROVED** [review]
Aight, I'm on board with using vcpkg to start out. It  should be straightforward to prototype adding any deps we need quickly with this solution.

**Comment by fgaz** [comment]
I maintain the [NixOS](https://nixos.org) package for Ladybird, and unfortunately this change makes my work a lot more difficult, because vcpkg always tries to download sqlite from the net instead of using the installed one. It can be tricked into believing sqlite was already downloaded, [but it's not pretty](https://github.com/NixOS/nixpkgs/blob/a73df7dc951cdc1dd7060340a524f0124cec4986/pkgs/applications/misc/trenchbroom/default.nix#L20-L64).

Switching to Find* scripts (or using them if vcpkg is not found) would work much better for us and for other distros.

**Comment by trflynn89** [comment]
If you look at the 65ddd0553b21289db60eb1e11d27ce6fa90eac15, you'll see we do use `find_package(SQLite3)`.

If you want to build without vcpkg, you can comment out this line:
https://github.com/LadybirdWebBrowser/ladybird/blob/65ddd0553b21289db60eb1e11d27ce6fa90eac15/CMakeLists.txt#L3

Or submit a PR here to add an option to disable vcpkg, and make that line conditional on that option.

**Comment by fgaz** [comment]
> If you look at the [65ddd05](https://github.com/LadybirdWebBrowser/ladybird/commit/65ddd0553b21289db60eb1e11d27ce6fa90eac15), you'll see we do use `find_package(SQLite3)`.



Oops, I totally missed that :sweat_smile:. 

> Or submit a PR here to add an option to disable vcpkg, and make that line conditional on that option.

Will do

**Comment by vitalyster** [comment]
This variable can be set from your build shell scripts, no need to introduce options and make CMakeLists.txt file overcomplicated

**Pull request merged**